File: Front/src/app/models/Reservation.Model.ts
export class ReservationModel{
    reservation_id: number;
    hotel_id: number;
    client_id: number;
    start_date: Date;
    end_date: Date;
    reserved_rooms: number;
    no_guests: number;



    constructor()
    {
        this.reservation_id = 0;
        this.hotel_id = 0 ;
        this.client_id = 0 ;
        this.start_date = new Date;
        this.end_date = new Date;
        this.reserved_rooms = 0;
        this.no_guests = 0;
    }
    
}

File: Back/src/main/java/com/example/booking/project/models/entities/ReservationModel.java
package com.example.booking.project.models.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name="Reservations")

public class ReservationModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long reservation_id;

    @Column
    private Long hotel_id;

    @Column
    private Long client_id;

    @Column
    private LocalDate start_date;

    @Column
    private LocalDate end_date;

    @Column
    private int reserved_rooms;

    @Column
    private int no_guests;


}
